### ISSUE TITLE
FR: tighter integration with occur

### ISSUE THREAD

**Issue by aspiers** [raxod502/ctrlf#69]
`M-s o` for `ctrlf-occur` is a really nice feature; however it doesn't make usage of occur very convenient when the ctrlf minibuffer is active:

- The resulting `*Occur*` buffer is not updated in any way as the user navigates through the ctrlf matches.  It would be cool if the point gets moved around the `*Occur*` buffer to correspond to the currently selected ctrl match.
- Once you hit `Enter`, the `*Occur*` buffer is immediately hidden, so if you want to (say) use `occur-edit-mode`, you have to unhide it again.  It would be cool if there was a way to exit without hiding it.

**Comment by raxod502** [comment]
Yes, those changes both seem reasonable, as long as the implementation does not turn out to be obtusely complicated for some unexpected reason.
